### DISCUSSION TITLE
How to set a unique id for a new entity?

### DISCUSSION THREAD

**Discussion by wsw70** [custom-components/pyscript#321] [Q&A] [answered]
I created a new entity from within my script:

    binary_sensor.test1 = "aaa"

I see it correctly in HA (with the correct value) but cannot edit it from within there because it has no unique ID:

![image](https://user-images.githubusercontent.com/1345886/153931072-ab798224-7ec2-4cf6-8620-c4cb7da9f565.png)

(*this entity has no unique ID and therefore its parameters cannot be set from the interface ...*) 

Is there a way I can set this unique ID when creating the entity? (probably by first checking for its state and if this throws an exception - set it)

**Comment by dlashua** [comment, marked as answer]
There is a way to do this, however, it is not something PyScript handles for you. Therefore, you will need to consult the HomeAssistant developer documentation and do it in the same way the you would if you were writing a Home Assistant Integration.

For this type of control, it will likely be easier for you to write an Integration than it will be for you to do it through PyScript.

**Reply by wsw70** [reply]
Thanks Daniel. It was more for aesthetic reasons that I wanted to have this - the entities will be generated dynamically and used as state machines, the visual parts will be handled by a specific card (if the need arises at all).

This said, I will have a look at the documentation because as HA is becoming my "hub for everything, not only home automation", i will be better off understanding some more advanced mechanisms.
